src/Compilers/Core/Portable/DiagnosticAnalyzer/UnresolvedAnalyzerReference.cs

```diff
@@ -36,9 +36,12 @@ public override string FullPath
             }
         }
 
-        public override bool IsUnresolved
+        public override string Id
         {
-            get { return true; }
+            get
+            {
+                return _unresolvedPath;
```

Comment:
Can this be null? If so, we should probably use the same approach as in AnalyzerImageReference and return a unique Guid string as fallback.